lib/replicache.dart

```diff
@@ -332,25 +346,31 @@ class Replicache implements ReadTransaction {
       return;
     }
 
-    if (_timer == null) {
-      // Another call stack is already inside sync();
+    if (_syncFuture != null) {
```

Comment:
This is clever, and cleaner, but why `await` the syncFuture? Seems like just exiting immediately (as it did before) would be fine too.

Reply:
I wanted to be able to await the call to `sync` too and now that the existing sync will be done when the call completes